lib/axon.ex

```diff
@@ -1355,6 +1355,62 @@ defmodule Axon do
     {{new_c, new_h}, output_sequence}
   end
 
+  @doc """
+  Freezes parameters returned from `fun` in the given
+  model. `fun` takes the model's parameter list and returns
+  the list of parameters it wishes to freeze.
```

Comment:
For people like me, who have no idea what freeze in the context of NN mean, what does this function actually do? :) Can you please add some examples too? Also, is it possible to test this? Other than that, code looks great! :heart: